view/readrom.js

```diff
@@ -4,55 +4,74 @@
  * SPDX-License-Identifier: Apache-2.0
  */
 
-function load_bin(file){
+/**
+ * 
+ * @param {*} dev 
+ * @param {*} file 
+ * @param {*} reader_method 
+ * @param {*} callback 
+ * @param {*} external_params 
+ * You can use variable `load_returns` to get the return value of `dev.loadFile`.
+ */
+function loadToDevice(dev, file, reader_method, loadfile_external_params=[], callback){
     let reader = new FileReader();
-    const isos = $("#os").prop("checked");
     reader.addEventListener('load', function(e) {
         let binary = e.target.result;
-        if (isos) {
-            zealcom.rom.loadFile(binary);
-            $("#binready").addClass("ready");
-        } else {
-            const addr = $("#address").val();
-            const result = parseInt(addr, 16);
-            zealcom.ram.loadFile(result, binary);
-        }
+        let load_returns = dev.loadFile(binary, ...loadfile_external_params);
```

Comment:
This may not work if the restart button is pressed. When the machine is restarted, all the components are destroyed and re-created but this listener won't be updated, so the address `dev` still points to the older device